project/Build.scala

```diff
@@ -85,6 +89,7 @@ object Shading extends Build {
     ),
     assemblyOption in assembly := (assemblyOption in assembly).value.copy(includeScala = false)
   ).settings(commonSettings: _*)
+  .enablePlugins(ProtobufPlugin)
```

Comment:
I believe Spark >= 2.2 still has a dependency on protobuf 2. I am not sure how the protobuf plugin will interact with the shading rules above, so I recommend you manually inspect the jar files produced to ensure that there is nothing in the `google.protobuf` and `com.google.protobuf` namespaces. That may be easy to add also to travis if you want to automate it: `unzip -l xxx.jar | grep google.protobuf | grep -e tensorframes | ...`

Reply:
We dropped Spark 2.2 support in #152. I checked the jar manually and it looks fine. Will add a test in a follow-up PR.